client-common/src/tendermint/http_rpc_client.rs

```diff
@@ -115,19 +147,21 @@ impl Client for RpcClient {
 
     fn block(&self, height: u64) -> Result<Block> {
         let params = [json!(height.to_string())];
-        self.call("block", &params)
+        Ok(self.call::<BlockResponse>("block", &params)?.block)
     }
 
     fn block_batch<'a, T: Iterator<Item = &'a u64>>(&self, heights: T) -> Result<Vec<Block>> {
-        let params = heights
+        let params: Vec<(&str, Vec<Value>)> = heights
             .map(|height| ("block", vec![json!(height.to_string())]))
-            .collect::<Vec<(&str, Vec<Value>)>>();
-        let response = self.call_batch::<Block>(&params)?;
+            .collect();
 
-        response
-            .into_iter()
-            .map(|block| block.chain(|| (ErrorKind::InvalidInput, "Block information not found")))
-            .collect::<Result<Vec<Block>>>()
+        let rsps = self.call_batch::<BlockResponse>(&params)?;
+        rsps.into_iter()
+            .map(|rsp| {
+                rsp.chain(|| (ErrorKind::InvalidInput, "Block information not found"))
+                    .map(|rsp_| rsp_.block)
```

Comment:
```suggestion
                    .map(|rsp| rsp.block)
```

Reply:
It's for avoiding the same name confusion with outer closure.

Reply:
Ok.